### ISSUE TITLE
Crash on Android with Xamari.Forms 4.5.0.282-pre4 Release builds while using DisplayActionSheet

### ISSUE THREAD

**Issue by vsfeedback** [xamarin/Xamarin.Forms#9696]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/926634/crash-on-android-with-xamariforms-450282-pre4-rele.html)._

---
<p>I have just noticed a bug in Xamarin.Forms 4.5.0.282-pre4 that was not present in 4.4.0.991640 when opening a Page.DisplayActionSheet on android it crashes. It only happens in release builds and not in Debug and doesn’t happen in IOS. It doesn’t seem to matter where it is called(I copied the code to an OnAppearing method in a different page and changing the number of items or the text has no effect on it.</p>
<pre>    private async void FilterToobarItem_Clicked(object sender, EventArgs e)
    {
        string action = await DisplayActionSheet(&quot;Filter&quot;, &quot;Cancel&quot;, null, &quot;All&quot;, &quot;Landmarks&quot;, &quot;Attractions&quot;, &quot;History&quot;, &quot;Famous People&quot;, &quot;Historical Buildings&quot;, &quot;Demolished Historical Buildings&quot;);</pre>
</pre>

---
#### Visual Studio Feedback System on 2/23/2020, 09:24 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by samhouts** [comment]
similar to #9031

**Comment by ichijikuibo** [comment]
It only happens when a code shrinker is enabled either proguard or r8 and I also tested it in a new Master Detail project and it also happens.

**Comment by jsuarezruiz** [comment]
Could it have been fixed with https://github.com/xamarin/Xamarin.Forms/pull/9318?

**Comment by jsuarezruiz** [comment]
I confirm the issue using Xamarin.Forms 4.5-pre4 and only happens if a code shrinker is enabled.

[Issue9696.zip](https://github.com/xamarin/Xamarin.Forms/files/4250734/Issue9696.zip)

```
[MonoDroid] UNHANDLED EXCEPTION:
[MonoDroid] System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> Java.Lang.RuntimeException: Binary XML file line #18: Binary XML file line #18: Error inflating class android.support.v7.widget.AlertDialogLayout ---> Java.Lang.RuntimeException: Binary XML file line #18: Error inflating class android.support.v7.widget.AlertDialogLayout ---> Java.Lang.ClassNotFoundException: Didn't find class "android.support.v7.widget.AlertDialogLayout" on path: DexPathList[[zip file "/data/app/com.companyname.issue9696-Umdw5MSHhRT_LWQMpDYVFQ==/base.apk"],nativeLibraryDirectories=[/data/app/com.companyname.issue9696-Umdw5MSHhRT_LWQMpDYVFQ==/lib/x86, /data/app/com.companyname.issue9696-Umdw5MSHhRT_LWQMpDYVFQ==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
[MonoD02-25 16:22:51.502 I]    --- End of inner exception stack trace ---
[MonoDroid]    --- End of inner exception stack trace ---
[MonoDroid]   at System.Reflection.MethodBase.02-25 16:22:51.502 I/MonoDroid(26601):   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0002a] in <87b4274105cd483cbc0dfad63bc848b3>:0 
[MonoDroid]   at Android.App.Dialog.Show () [0x0000a] in <086f830e45834f109f354d6517cd41b5>:0 
[MonoDroid]   at Xamarin.Forms.Platform.Android.PopupManager+PopupRequestHelper+FlexibleAlertDialog.Show () [0x00008] in <e4bdfd66d0ca4b71bbba949c4323087e>:0 
[MonoDroid]   at Xamaris arguments) [0x000dc] in <e4bdfd66d0ca4b71bbba949c4323087e>:0 
[MonoDroid]   at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
[MonoDroid]   at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <94a45be22dca481bada535729ee07660>:0 
[MonoDroid]    --- End of inner exception stack trace ---
[MonoDroid]   at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00086] in <94a45be22dca481bada535729ee07660>:0 
[MonoDroid]   at Xamarin.Forms.MessagingCenter+Subscription.InvokeCallback (System.Object sender, System.Object args) [0x00076] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[MonoDroid]   at Xamarin.Forms.MessagingCenter.InnerSend (System.String message, System.Type senderType, System.Type argType, System.Object sender, System.Object args) [0x0006b] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[MonoDroid]   at Xamarin.Forms.MessagingCenter.Xamarin.Forms.IMessagingCenter.Send[TSender,TArgs] (TSender sender, System.String message, TArgs args) [0x00013] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[MonoDroid]   at Xamarin.Forms.MessagingCenter.Send[TSender,TArgs] (TSender sender, System.String message, TArgs args) [0x00005] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[MonoDroid]   at Xamarin.Forms.Page.DisplayActionSheet (System.String title, System.String cancel, System.String destruction, System.String[] buttons) [0x00025] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[MonoDroid]   at Issue9696.MainPage.Button_Clicked (System.Object sender, System.EventArgs e) [0x00011] in <82512ae61148473b9be2521a394be0a3>:0 
[MonoDroid]   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <94a45be22dca481bada535729ee07660>:0 
[MonoDroid]   at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <086f830e45834f109f354d6517cd41b5>:0 
[MonoDroid]   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <086f830e45834f109f354d6517cd41b5>:0 
[MonoDroid]   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <086f830e45834f109f354d6517cd41b5>:0 
[MonoDroid]   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.42(intptr,intptr)
[zygote] JNI RegisterNativeMethods: attempt to register 0 native methods for android.runtime.JavaProxyThrowable
[AndroidRuntime] Shutting down VM
[AndroidRuntime] FATAL EXCEPTION: main
[AndroidRuntime] Process: com.companyname.issue9696, PID: 26601
[AndroidRuntime]    --- End of inner exception stack trace ---
[AndroidRuntime]    --- End of inner exception stack trace ---
[AndroidRuntime]   at Java.Interop.JniEnvironment+InstanceMethods.CallVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x0006e] in <87b4274105cd483cbc0dfad63bc848b3>:0 
[AndroidRuntime]   at Xamarin.Forms.Platform.Android.PopupManager+PopupRequestHelper+FlexibleAlertDialog.Show () [0x00008] in <e4bdfd66d002-25 16:22:51.513 E/AndroidRuntime(26601):   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0002a] in <87b4274105cd483cbc0dfad63bc848b3>:0 
[AndroidRuntime]   at Android.App.Dialog.Show () [0x0000a] in <086f830e45834f109f354d6517cd41b5>:0 
[AndroidRuntime]   at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflectionca4b71bbba949c4323087e>:0 
[AndroidRuntime]   at Xamarin.Forms.Platform.Android.PopupManager+PopupRequestHelper.OnActionSheetRequested (Xamarin.Forms.Page sender, Xamarin.Forms.Internals.ActionSheetArguments arguments) [0x000dc] in <e4bdfd66d0ca4b71bbba949c4323087e>:0 
[AndroidRuntime]   at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
[AndroidRuntime]   at Xamarin.Forms.MessagingCenter.Xamarin.Forms.IMessagingCenter.Send[TSender,TArgs] (TSender sender, System.String message, TArgs args) [0x00013] in <04ebaa15464602-25 16:22:51.513 E/AndroidRuntime(26601):   at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <94a45be22dca481bada535729ee07660>:0 
[AndroidRuntime]    --- End of inner exception stack trace ---
[AndroidRuntime]   at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <94a45be22dca481bada535729ee07660>:0 
[AndroidRuntime]   at Xamarin.Forms.MessagingCenter+Subscription.InvokeCallback (System.Object sender, System.Object args) [0x00076] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[AndroidRuntime]   at Xamarin.Forms.MessagingCenter.InnerSend (System.String message, System.Type senderType, System.Type argType, System.Object sender, System.Object args) [0x0006b] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[AndroidRuntime] 	at android.app.ActivityThread.main(Ac02-25 16:22:51.513 E/AndroidRuntime(26601):   at Xamarin.Forms.MessagingCenter.Xamarin.Forms.IMessagingCenter.Send[TSender,TArgs] (TSender sender, System.String message, TArgs args) [0x00013] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[AndroidRuntime]   at Xamarin.Forms.MessagingCenter.Send[TSender,TArgs] (TSender sender, System.String message, TArgs args) [0x00005] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[AndroidRuntime]   at Xamarin.Forms.Page.DisplayActionSheet (System.String title, System.String cancel, System.String destruction, System.String[] buttons) [0x00025] in <04ebaa154646441fa0dfe2a6b372f39f>:0 
[AndroidRuntime]   at Issue9696.MainPage.Button_Clicked (System.Object sender, System.EventArgs e) [0x00011] in <82512ae61148473b9be2521a394be0a3>:0 
[AndroidRuntime]   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <94a45be22dca481bada535729ee07660>:0 
[AndroidRuntime]   at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <086f830e45834f109f354d6517cd41b5>:0 
[AndroidRuntime]   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <086f830e45834f109f354d6517cd41b5>:0 
[AndroidRuntime]   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <086f830e45834f109f354d6517cd41b5>:0 
[AndroidRuntime]   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.42(intptr,intptr)
[AndroidRuntime] 	at mono.java.lang.RunnableImplementor.n_run(Native Method)
[AndroidRuntime] 	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
[AndroidRuntime] 	at android.os.Handler.handleCallback(Handler.java:790)
[AndroidRuntime] 	at android.os.Handler.dispatchMessage(Handler.java:99)
[AndroidRuntime] 	at android.os.Looper.loop(Looper.java:164)
[AndroidRuntime] 	at android.app.ActivityThread.main(ActivityThread.java:6494)
[AndroidRuntime] 	at java.lang.reflect.Method.invoke(Native Method)
[AndroidRuntime] 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
[AndroidRuntime] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by ichijikuibo** [comment]
This problem is fixed now with the latest release 4.5.0.356

![DisplayActionSheet](https://user-images.githubusercontent.com/61333434/75467998-6992b980-5984-11ea-8b06-e151e3da19af.png)

**Comment by jfversluis** [comment]
Since this is reported fixed, I will close this for now.
